src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 Google LLC
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     https://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.*;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.RepositoryType;
+import com.google.sps.user.User;
+import com.google.sps.user.repository.UserRepository;
+import com.google.sps.user.repository.UserRepositoryFactory;
+import org.jetbrains.annotations.Nullable;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This servlet handles users' data.
+ */
+@WebServlet("/user-data-servlet")
+public class UserDataServlet extends HttpServlet {
+  public static final String NAME_INPUT = "name";
+  public static final String PUBLIC_PORTFOLIO_INPUT = "publicPortfolio";
+  public static final String PUBLIC_PORTFOLIO_INPUT_PUBLIC_VALUE = "public";
+  public static final String SELF_INTRODUCTION_INPUT = "selfIntroduction";
+  public static final String IMG_KEY_INPUT = "imgKey";
+  public static final String DELETE_IMG_INPUT = "deleteImg";
+
+  private final UserRepository userRepository =
+          UserRepositoryFactory.getUserRepository(RepositoryType.DATASTORE);;
+  private final UserService userService = UserServiceFactory.getUserService();;
+
+  private final BlobstoreService blobstoreService;
+  private final BlobInfoFactory blobInfoFactory;
+
+  /**
+   * For production.
+   */
+  public UserDataServlet() {
+    blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    blobInfoFactory = new BlobInfoFactory();
```

Comment:
this(BlobstoreServiceFactory.getBlobstoreService(), new BlobInfoFactory());

so we don't repeat code.

Reply:
Done.